clarity/contracts/arkadiko-alex-dual-yield-v1-1.clar

```diff
@@ -0,0 +1,74 @@
+;; @contract ALEX Dual Yield - Mints DIKO for ALEX dual yield efforts
+;; @version 1.1
+
+;; Errors
+(define-constant ERR-NOT-AUTHORIZED u32401)
+
+;; Basic emission percentage (5.5%)
+(define-data-var rewards-percentage uint u55000)
+(define-data-var shutdown-activated bool false)
+
+(define-read-only (is-activated)
+  (and
+    (is-eq (unwrap-panic (contract-call? .arkadiko-dao get-emergency-shutdown-activated)) false)
+    (is-eq (var-get shutdown-activated) false)
+  )
+)
+
+(define-read-only (get-rewards-per-block-for-alex)
+  (let (
+    (total-staking-rewards (contract-call? .arkadiko-diko-guardian-v1-1 get-staking-rewards-per-block))
+    (pool-percentage (var-get rewards-percentage))
+  )
+    (if (is-activated)
+      (/ (* total-staking-rewards pool-percentage) u1000000)
+      u0
+    )
+  )
+)
+
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; public methods (callable by ALEX contract or Arkadiko DAO) ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define-public (mint (amount-of-blocks uint))
```

Comment:
Not sure, but I think Alex needs a mint method like:
`(define-public (mint (amount uint) (recipient principal))`

In the mint method we should:
- Mint DIKO and assign to contract (diko-to-mint = blocks-passed * get-rewards-per-block-for-alex)
- Send DIKO based on amount and recipient parameters

This way we are in full control of minted DIKO.

Reply:
@fiftyeightandeight I will let you comment on this. Let us know what you would need exactly or if this works

Reply:
`(amount uint) (recipient principal)` would definitely help. Thank you.

Reply:
@fiftyeightandeight ok I will look at it and refactor!

Reply:
Hi @nieldeckx we need to pre-agree to the exact exact emission schedule, which will then be hard-coded into the relevant contracts.

Reply:
@fiftyeightandeight I think what Niel means is that 1.25M DIKO vs 5.5% of our emissions are not exactly the same (could differ with a few thousand DIKO)... so to get exactly 1.25M DIKO would be very hard for us to calculate as we approximate it with a percentage and then there is a "rounding error" of a few basis points

I think what you could do on your end is call `get-rewards-per-block-for-alex` * `number-of-blocks` to get the exact amount of DIKO for the staking period you're giving out rewards for?

Reply:
Ok, that should work too. So `mint` will accept, perhaps, `(amount-of-blocks uint)` and `(recipient principal)` as input?

Reply:
Actually, `mint` will mint all the rewards for the number of blocks, correct? I.e. If the number of blocks was 525 blocks, then calling `mint` will mint 525 x 5.5% of your emission?

Reply:
> Not sure, but I think Alex needs a mint method like: `(define-public (mint (amount uint) (recipient principal))`
> 
> In the mint method we should:
> 
> * Mint DIKO and assign to contract (diko-to-mint = blocks-passed * get-rewards-per-block-for-alex)
> * Send DIKO based on amount and recipient parameters
> 
> This way we are in full control of minted DIKO.

If you provide implementation like the above, think we can handle the amount to send to each `recipient` based on total DIKO minted to the contract up to that point.

Reply:
@fiftyeightandeight yeah, ok.. we are going to simplify things a bit on our side.

We'll make a method `(define-public (mint (amount uint) (recipient principal))`. We have 1.25M DIKO per year and approximately 100 ALEX cycles per year (144 * 365 / 525 = 100.114 cycles). We will just mint 1.25M DIKO on the contract (not all at once but through a mint function we can call) and you can assume you can emit 12,500 DIKO per cycle (1.25M / 100). This way you can just pro-rata call the `mint` function per recipient.

Would that be fine? Sorry for the back and forth but I think this will be the easiest where both sides have to do the least calculations.